src/connection.ts

```diff
@@ -979,6 +980,26 @@ class Connection extends EventEmitter {
     this.curTransientRetryCount = 0;
     this.transientErrorLookup = new TransientErrorLookup();
 
+    this.state = this.STATE.INITIALIZED;
+
+    process.nextTick(() => {
+      if (this.state === this.STATE.INITIALIZED) {
+        const message = 'In the next major version of `tedious`, creating a new ' +
+          '`Connection` instance will no longer establish a connection to the ' +
+          'server automatically. Please use the new `connect` helper function or ' +
+          'call the `.connect` method on the newly created `Connection` object to ' +
+          'silence this message.';
+        deprecate(message);
+        this.connect();
+      }
+    });
```

Comment:
This is a neat "trick" - if `new Connection()` is called without a call to `.connect` in the same event loop tick, `.connect` will be called automatically, but there will also be a deprecation message emitted. This ensures existing code continues working as-is.

The deprecation message can easily be silenced by calling `.connect` right after calling `new Connection()`, allowing users to migrate their code to the new API before upgrading to the next major version.